### PR TITLE
Add note about .NET8.0 metrics in readme

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #
Design discussion issue #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 1]
```diff
@@ -116,6 +116,19 @@ Http Semantic Conventions using `OTEL_SEMCONV_STABILITY_OPT_IN`.
 * If user sets the environment variable to `http/dup`, the instrumentation
   emits both `http.server.duration` and `http.server.request.duration`.
 
+##### Metrics on ASP.NET Core 8.0 and above frameworks
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:119]: Probably remove the term `frameworks`?
```suggestion
##### Metrics on ASP.NET Core 8.0 and above
```

File: src/OpenTelemetry.Instrumentation.Http/README.md [hunk 2]
```diff
@@ -125,6 +125,19 @@ Http Semantic Conventions using `OTEL_SEMCONV_STABILITY_OPT_IN`.
 * If user sets the environment variable to `http/dup`, the instrumentation
   emits both `http.client.duration` and `http.client.request.duration`.
 
+##### Metrics on .NET8.0 and above frameworks
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Http/README.md:128]:
```suggestion
##### Metrics on .NET8.0 and above
```

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 3]
```diff
@@ -116,6 +116,19 @@ Http Semantic Conventions using `OTEL_SEMCONV_STABILITY_OPT_IN`.
 * If user sets the environment variable to `http/dup`, the instrumentation
   emits both `http.server.duration` and `http.server.request.duration`.
 
+##### Metrics on ASP.NET Core 8.0 and above
+
+This library enables all [built-in
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:121]: `OTEL_SEMCONV_STABILITY_OPT_IN` -- is this respected for .net8?

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/README.md:121]: for tracing it is. not for metrics. Planning to remove for tracing too before next release.

Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.AspNetCore/README.md:121]: i meant to say - its not clear from docs, what is expected behavior in .NET8+newer for metrics.

We could write that .NET8+, AddInst() is simply calling AddMeter(meternames), and nothing more. And explicitly state that no other feature of this package, like the semconv_opt-In is respected in .NET8+.

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/README.md:121]: Once we remove semconv_opt-In (which we will [soon](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5066)), there is no additional feature that instrumentation library will provide which will be available in net6.0/net7.0 but not in net8.0. 

> AddInst() is simply calling AddMeter(meternames)

This is an implementation detail, not sure if we need to mention this here.

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/README.md:121]: I plan to rewrite this section(in follow up) as well: https://github.com/open-telemetry/opentelemetry-dotnet/blob/8ee44877077d9c44154513a1fd4e1d06fd0b43d5/src/OpenTelemetry.Instrumentation.AspNetCore/README.md#list-of-metrics-produced. That will make it clear what is the difference between lower versions and net8.0.

**Review by Yun-Ting — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 4]
```diff
@@ -90,31 +90,51 @@ public void ConfigureServices(IServiceCollection services)
...
+the following metric:
+
+| Name                              | Details                                                                                                       |
+|-----------------------------------|---------------------------------------------------------------------------------------------------------------|
+| `http.server.request.duration`    | [Specification](https://github.com/open-telemetry/semantic-conventions/blob/release/v1.23.x/docs/http/http-metrics.md#metric-httpserverrequestduration) |
+
+Starting from `.NET8.0`, metrics instrumentation is natively implemented, and
+the ASP.NET Core library has incorporated support for [built-in
+metrics](https://learn.microsoft.com/dotnet/core/diagnostics/built-in-metrics-aspnetcore)
+following the OpenTelemetry semantic conventions. The library include additional
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:103]:
```suggestion
following the OpenTelemetry semantic conventions. The library includes additional
```

File: src/OpenTelemetry.Instrumentation.Http/README.md [hunk 5]
```diff
@@ -99,31 +99,51 @@ to see how to enable this instrumentation in an ASP.NET application.
...
+instrumentation emits the following metric:
+
+| Name                              | Details                                                                                                       |
+|-----------------------------------|---------------------------------------------------------------------------------------------------------------|
+| `http.client.request.duration`    | [Specification](https://github.com/open-telemetry/semantic-conventions/blob/release/v1.23.x/docs/http/http-metrics.md#metric-httpclientrequestduration) |
+
+Starting from `.NET8.0`, metrics instrumentation is natively implemented, and
+the HttpClient library has incorporated support for [built-in
+metrics](https://learn.microsoft.com/dotnet/core/diagnostics/built-in-metrics-system-net)
+following the OpenTelemetry semantic conventions. The library include additional
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Http/README.md:112]:
```suggestion
following the OpenTelemetry semantic conventions. The library includes additional
```

File: src/OpenTelemetry.Instrumentation.Http/README.md [hunk 6]
```diff
@@ -99,31 +99,51 @@ to see how to enable this instrumentation in an ASP.NET application.
...
+
+**Note:** There is no difference in features or emitted metrics when enabling
+metrics using `AddMeter()` or `AddHttpClientInstrumentation()` on `.NET8.0` and
+newer versions.
+
+> **Note**
+> The `http.client.request.duration` metric is emitted in `seconds` as
+    per the semantic convention. While the convention [recommends using custom
+    histogram
+    buckets](https://github.com/open-telemetry/semantic-conventions/blob/2bad9afad58fbd6b33cc683d1ad1f006e35e4a5d/docs/http/http-metrics.md)
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Http/README.md:141]:
```suggestion
    buckets](https://github.com/open-telemetry/semantic-conventions/blob/release/v1.23.x/docs/http/http-metrics.md)
```

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 7]
```diff
@@ -90,31 +90,51 @@ public void ConfigureServices(IServiceCollection services)
...
-
-    This metric is emitted in `seconds` as per the semantic convention. While
-    the convention [recommends using custom histogram buckets](https://github.com/open-telemetry/semantic-conventions/blob/2bad9afad58fbd6b33cc683d1ad1f006e35e4a5d/docs/http/http-metrics.md)
-    , this feature is not yet available via .NET Metrics API.
-    A [workaround](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4820)
+When the application targets `.NET6.0` or `.NET7.0`, the instrumentation emits
+the following metric:
+
+| Name                              | Details                                                                                                       |
+|-----------------------------------|---------------------------------------------------------------------------------------------------------------|
```
Comment (Yun-Ting) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:97]:
```suggestion
|----|----|
```

**Review by utpilla — APPROVED** [review]

**Pull request merged**